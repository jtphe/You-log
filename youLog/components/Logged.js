import React from "react";
import { StyleSheet, Text, View, Image, TouchableOpacity } from "react-native";
import { Button } from "react-native-paper";
import ModalPassword from "./ModalPassword";
// import * as SQLite from "expo-sqlite";

/**
 * The Logged class
 * @reactProps {object} navigation - Navigation props
 */
class Logged extends React.Component {
  /**
   * Pass the title option to the navigation
   */
  static navigationOptions = {
    title: "YouLog"
  };

  /**
   * visibility : The visibility of the update password button | 
   * password:  The user password |
   * canUpdatePassword: If the user can update the password or not
   * @type {{visible: boolean, password: string, id: string, canUpdatePassword: boolean}}
   */
  state = {
    visible: false,
    password: '',
    id: null,
    canUpdatePassword: true,
    email: '',
    name: ''
  };

  UNSAFE_componentWillReceiveProps(){
    if(this.props.navigation.state.params.id){
      this.setState({
        password: JSON.stringify(
          this.props.navigation.state.params.password
        ).replace(/\"/g, "")
      });
    }
    if(this.props.navigation.state.params.id){
      this.setState({
        id: JSON.stringify(this.props.navigation.state.params.id).replace(
          /\"/g,
          ""
        )
      })
    }
    if( this.props.navigation.state.params){
      this.setState({
        email: this.props.navigation.state.params.email,
        name: this.props.navigation.state.params.name
      })
    }
  }

  /**
   * Update the password in the SQLite DB
   * @param {string} text - The new password
   */
  _updatePassword = text => {
    if (text !== this.state.password) {
      const usersDB = SQLite.openDatabase("users.db");
      usersDB.transaction(tx => {
        tx.executeSql(
          "CREATE TABLE IF NOT EXISTS users (id integer primary key not null, name text, email text, password text);"
        );
      });

      const query = "UPDATE users SET password = ? WHERE id = ?";
      const params = [text, this.state.id];

      usersDB.transaction(tx => {
        tx.executeSql(query, params);
      });
      this.setState({
        visible: false,
        canUpdatePassword: false
      });
    } else {
      this.setState({
        visible: false
      });
    }
  };

  /**
   * Hide the modal to change the password when user click on the cancel button
   */
  _cancel = () => {
    this.setState({
      visible: false
    });
  };

  /**
   * Show the modal to change the password
   */
  _showModal = () => {
    this.setState({
      visible: true
    });
  };

  /**
   * Convert the item from the sign in screen to json object
   * @param {string} params - The item to convert
   */
  _convertToJsObject = params => {
    return JSON.stringify(params).replace(/\"/g, "");
  };

  /**
   * Render the main screen
   * @returns {React.Component} - Logged Component
   */
  render() {
    const { visible, password, canUpdatePassword, email, name } = this.state;
    const EMAIL = this._convertToJsObject(email);
    const NAME = this._convertToJsObject(name);

    return (
      <View style={styles.container}>
        <Image
          style={styles.congratulation}
          source={require("../media/congratulation.png")}
        />
        <Text style={styles.welcomeTitle}>Bienvenue {NAME} sur YouLog üéâ</Text>
        <Text style={styles.welcomeMail}>
          Vous √™tes connect√© sous {EMAIL} ‚úâÔ∏è
        </Text>
        <Button
          mode={"contained"}
          labelStyle={{ color: "#4834d4" }}
          style={styles.btnSignUp}
          onPress={() => this.props.navigation.popToTop()}
        >
          D√©connexion
        </Button>
        {canUpdatePassword ? (
          <TouchableOpacity onPress={this._showModal}>
            <Text style={styles.password}>Changer votre mot de passe</Text>
          </TouchableOpacity>
        ) : null}
        <ModalPassword
          visible={visible}
          updatePassword={text => this._updatePassword(text)}
          cancel={this._cancel}
          password={password}
        />
      </View>
    );
  }
}

/**
 * Styles of the Logged Component
 */
const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: "column",
    alignItems: "center",
    justifyContent: "center"
  },
  btnSignUp: {
    backgroundColor: "#fff",
    padding: 8,
    paddingLeft: 35,
    paddingRight: 35,
    marginLeft: 50,
    marginRight: 50
  },
  welcomeTitle: {
    textAlign: "center",
    marginTop: 20,
    marginBottom: 20,
    marginLeft: 30,
    marginRight: 30,
    fontSize: 20
  },
  welcomeMail: {
    textAlign: "center",
    marginTop: 20,
    marginBottom: 40,
    marginLeft: 30,
    marginRight: 30,
    fontSize: 14
  },
  congratulation: {
    width: 180,
    height: 180,
    marginBottom: 20
  },
  password: {
    marginTop: 20,
    color: "#4834d4",
    textDecorationLine: "underline"
  }
});

export default Logged;
