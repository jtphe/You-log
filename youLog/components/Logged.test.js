import React from 'react';
import Logged from './Logged';

import renderer from 'react-test-renderer';

test('Logged test', ()=> {
    const tree = renderer.create(<Logged/>).toJSON();
    expect(tree).toMatchSnapshot();
})