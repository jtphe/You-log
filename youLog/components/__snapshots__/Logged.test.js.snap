// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logged test 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../media/congratulation.png",
      }
    }
    style={
      Object {
        "height": 180,
        "marginBottom": 20,
        "width": 180,
      }
    }
  />
  <Text
    style={
      Object {
        "fontSize": 20,
        "marginBottom": 20,
        "marginLeft": 30,
        "marginRight": 30,
        "marginTop": 20,
        "textAlign": "center",
      }
    }
  >
    Bienvenue 
    
     sur YouLog 🎉
  </Text>
  <Text
    style={
      Object {
        "fontSize": 14,
        "marginBottom": 40,
        "marginLeft": 30,
        "marginRight": 30,
        "marginTop": 20,
        "textAlign": "center",
      }
    }
  >
    Vous êtes connecté sous 
    
     ✉️
  </Text>
  <View
    style={
      Object {
        "backgroundColor": "#fff",
        "borderColor": "transparent",
        "borderRadius": 4,
        "borderStyle": "solid",
        "borderWidth": 0,
        "elevation": 2,
        "marginLeft": 50,
        "marginRight": 50,
        "minWidth": 64,
        "padding": 8,
        "paddingLeft": 35,
        "paddingRight": 35,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 0.75,
          "width": 0,
        },
        "shadowOpacity": 0.24,
        "shadowRadius": 1.5,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityStates={Array []}
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 4,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            },
            undefined,
          ]
        }
      >
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
                "textAlign": "left",
              },
              Array [
                Object {
                  "letterSpacing": 1,
                  "marginHorizontal": 16,
                  "marginVertical": 9,
                  "textAlign": "center",
                },
                undefined,
                Object {
                  "textTransform": "uppercase",
                },
                Object {
                  "color": "#ffffff",
                  "fontFamily": "System",
                  "fontWeight": "500",
                },
                Object {
                  "fontFamily": "System",
                  "fontWeight": "500",
                },
                Object {
                  "color": "#4834d4",
                },
              ],
            ]
          }
        >
          Déconnexion
        </Text>
      </View>
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#4834d4",
          "marginTop": 20,
          "textDecorationLine": "underline",
        }
      }
    >
      Changer votre mot de passe
    </Text>
  </View>
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={false}
    visible={false}
  >
    <View
      style={
        Object {
          "alignItems": "stretch",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 20,
            "marginBottom": 10,
            "textAlign": "center",
          }
        }
      >
        Modifier votre mot de passe
      </Text>
      <View
        style={
          Object {
            "marginTop": 20,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgb(231, 231, 231)",
                "borderTopLeftRadius": 4,
                "borderTopRightRadius": 4,
              },
              Object {
                "borderColor": "#ecf0f1",
                "borderRadius": 6,
                "borderWidth": 1,
                "marginBottom": 25,
                "marginLeft": 50,
                "marginRight": 50,
                "paddingLeft": 15,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.26)",
                "bottom": 0,
                "height": 2,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "transform": Array [
                  Object {
                    "scaleY": 0.5,
                  },
                ],
              }
            }
          />
          <View
            style={
              Object {
                "minHeight": 50,
                "paddingBottom": 0,
                "paddingTop": 0,
              }
            }
          >
            <TextInput
              allowFontScaling={true}
              autoCapitalize="none"
              editable={true}
              multiline={false}
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              placeholderTextColor="rgba(0, 0, 0, 0.54)"
              rejectResponderTermination={true}
              selectionColor="#6200ee"
              style={
                Array [
                  Object {
                    "flexGrow": 1,
                    "margin": 0,
                    "textAlign": "left",
                    "zIndex": 1,
                  },
                  Object {
                    "paddingHorizontal": 12,
                  },
                  Object {
                    "height": 50,
                  },
                  Object {
                    "paddingBottom": 0,
                    "paddingTop": 0,
                  },
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": undefined,
                    "textAlignVertical": "center",
                  },
                ]
              }
              underlineColorAndroid="transparent"
              value=""
            />
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#4834d4",
            "borderColor": "transparent",
            "borderRadius": 4,
            "borderStyle": "solid",
            "borderWidth": 0,
            "elevation": 2,
            "marginLeft": 50,
            "marginRight": 50,
            "minWidth": 64,
            "padding": 8,
            "paddingLeft": 35,
            "paddingRight": 35,
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 0.75,
              "width": 0,
            },
            "shadowOpacity": 0.24,
            "shadowRadius": 1.5,
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityStates={Array []}
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 4,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "textAlign": "left",
                  },
                  Array [
                    Object {
                      "letterSpacing": 1,
                      "marginHorizontal": 16,
                      "marginVertical": 9,
                      "textAlign": "center",
                    },
                    undefined,
                    Object {
                      "textTransform": "uppercase",
                    },
                    Object {
                      "color": "#ffffff",
                      "fontFamily": "System",
                      "fontWeight": "500",
                    },
                    Object {
                      "fontFamily": "System",
                      "fontWeight": "500",
                    },
                    Object {
                      "color": "#fff",
                    },
                  ],
                ]
              }
            >
              Annuler
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#fff",
            "borderColor": "transparent",
            "borderRadius": 4,
            "borderStyle": "solid",
            "borderWidth": 0,
            "elevation": 2,
            "marginLeft": 50,
            "marginRight": 50,
            "marginTop": 20,
            "minWidth": 64,
            "padding": 8,
            "paddingLeft": 35,
            "paddingRight": 35,
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 0.75,
              "width": 0,
            },
            "shadowOpacity": 0.24,
            "shadowRadius": 1.5,
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityStates={Array []}
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 4,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "textAlign": "left",
                  },
                  Array [
                    Object {
                      "letterSpacing": 1,
                      "marginHorizontal": 16,
                      "marginVertical": 9,
                      "textAlign": "center",
                    },
                    undefined,
                    Object {
                      "textTransform": "uppercase",
                    },
                    Object {
                      "color": "#ffffff",
                      "fontFamily": "System",
                      "fontWeight": "500",
                    },
                    Object {
                      "fontFamily": "System",
                      "fontWeight": "500",
                    },
                    Object {
                      "color": "#4834d4",
                    },
                  ],
                ]
              }
            >
              Valider
            </Text>
          </View>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;
